src/redux/reducers/inputReducer.js

```diff
@@ -31,6 +37,41 @@ export default function inputReducer(state = initialState, action) {
         ),
         tasksInProgress: state.tasksInProgress.concat([action.payload]),
       };
+
+    case MOVE_TASK_IN_TO_DO:
+      return {
+        ...state,
+        tasksInProgress: state.tasksInProgress.filter(
+          (task) => task.id !== action.payload.id
+        ),
+        tasksToDo: state.tasksToDo.concat([action.payload]),
+      };
+
+    case MOVE_TASK_TO_DONE:
+      return {
+        ...state,
+        tasksInProgress: state.tasksInProgress.filter(
+          (task) => task.id !== action.payload.id
+        ),
+        tasksDone: state.tasksDone.concat([action.payload]),
+      };
+
+    case MOVE_TASK_TO_IN_PROGRESS_FROM_DONE:
+      return {
+        ...state,
+        tasksDone: state.tasksDone.filter(
+          (task) => task.id !== action.payload.id
+        ),
+        tasksInProgress: state.tasksInProgress.concat([action.payload]),
+      };
+
+    case DELETE_TASK_FROM_DONE:
+      return {
+        ...state,
+        tasksDone: state.tasksDone.filter(
+          (task) => task.id !== action.payload.id
+        ),
```

Comment:
create function to reuse `state.tasksDone.filter(
          (task) => task.id !== action.payload.id
        )`
 kinda function filterList(list, taskId): filteredList

Reply:
fixed